src/osp/scientific/kepler.cpp

```diff
@@ -0,0 +1,368 @@
+/**
+ * Open Space Program
+ * Copyright © 2019-2024 Open Space Program Project
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "kepler.h"
+#include <algorithm>
+#include <cassert>
+#include <cmath>
+
+using Magnum::Math::cross;
+using Magnum::Math::dot;
+
+namespace {
+
+/**
+ * @brief Wrap an angle to the range [0, 2π), useful for keplerian orbits
+ */
+static constexpr double wrap_angle(double const angle)
+{
+    double wrapped = std::fmod(angle, 2.0 * M_PI);
+    if (wrapped < 0)
+    {
+        wrapped += 2.0 * M_PI;
+    }
+    return wrapped;
+}
+
+static double solve_kepler_elliptic(double const meanAnomaly, double const eccentricity, int steps)
+{
+    double const tolerance = 1.0E-10;
+
+    // If eccentricity is low, then this is a circular orbit, so M = E
+    if (std::abs(eccentricity) < 1.0E-9)
+    {
+        return meanAnomaly;
+    }
+
+    double delta;
+
+    // initial guess
+    double E = meanAnomaly;
+
+    // use newton's method
+    // todo: implement something which converges more quickly
+    do
+    {
+        double f_E = E - eccentricity * std::sin(E) - meanAnomaly;
+        double df_E = 1.0 - eccentricity * std::cos(E);
+
+        delta = f_E / df_E;
+        E = E - delta;
+
+        E = wrap_angle(E); // Sometimes this sometimes fails to converge without this line
+        --steps;
+    } while (std::abs(delta) > tolerance && steps > 1);
+    return E;
+}
+
+static double solve_kepler_hyperbolic(double const meanAnomaly, double const eccentricity, int steps)
```

Comment:
can you add some doxygen documentation?

don't go overboard. Just a one sentence explanation of the point of the function is enough.